src/Domain/LeanCode.DomainModels.EF/CachingEFRepository.cs

```diff
@@ -0,0 +1,67 @@
+using System.Diagnostics.CodeAnalysis;
+using LeanCode.DomainModels.Model;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Internal;
+
+namespace LeanCode.DomainModels.EF;
+
+public abstract class CachingEFRepository<TEntity, TIdentity, TContext> : EFRepository<TEntity, TIdentity, TContext>
+    where TEntity : class, IAggregateRootWithoutOptimisticConcurrency<TIdentity>
+    where TIdentity : notnull
+    where TContext : DbContext
+{
+    protected CachingEFRepository(TContext dbContext)
+        : base(dbContext) { }
+
+    /// <summary>
+    /// Finds an entity by primary key. If the entity with provided key is tracked by the underlying
+    /// <see cref="DbContext" />, this method will return the cached entity.
+    /// </summary>
+    /// <remarks>For implementers: the default implementation won't work for composite primary keys.</remarks>
+    /// <param name="id">The identifier that</param>
+    /// <param name="cancellationToken">A <see cref="CancellationToken" /> to observe while waiting for the task to complete.</param>
+    /// <returns>Task with the found entity, or -null- if not found.</returns>
+    public override async Task<TEntity?> FindAsync(TIdentity id, CancellationToken cancellationToken = default)
+    {
+        return await FindTrackedOrLoadNewAsync(
```

Comment:
Nit: why `async`/`await`? `AsTask` should be enough.

Reply:
Stacktrace. `FindTrackedOrLoadNewAsync` is internal, so if someone forgets to override it for composite key (or anything bad happens), they will get be rather lost on what happened.